conans/client/build/cmake_flags.py

```diff
@@ -294,12 +294,18 @@ def get_definitions(self):
         definitions.update(build_type_definition(self._forced_build_type, build_type,
                                                  self._generator, self._output))
 
-        if tools.is_apple_os(os_):
-            definitions["CMAKE_OSX_ARCHITECTURES"] = tools.to_apple_arch(arch)
+        # don't attempt to override variables set within toolchain
+        if tools.is_apple_os(os_) and "CONAN_CMAKE_TOOLCHAIN_FILE" not in os.environ:
```

Comment:
is this reliable enough, or should I also check for `CMAKE_TOOLCHAIN_FILE in definitions` just in case?

Reply:
Yeah, better check both.